libvast/src/segment.cpp

```diff
@@ -80,19 +86,25 @@ chunk_ptr segment::chunk() const {
   return chunk_;
 }
 
-caf::expected<std::vector<table_slice_ptr>>
+caf::expected<std::vector<table_slice>>
 segment::lookup(const vast::ids& xs) const {
-  std::vector<table_slice_ptr> result;
-  auto f = [](auto buffer) {
-    auto slice = buffer->data_nested_root();
-    return std::pair{slice->offset(), slice->offset() + slice->rows()};
+  std::vector<table_slice> result;
+  auto f = [=](auto buffer) {
+    auto slice = table_slice{};
+    if (auto err = unpack(*buffer, slice)) {
+      VAST_ERROR(this, "failed to unpack table slice");
+      return std::pair{invalid_id, invalid_id};
+    }
+    // FIXME(mavam): get offset here
+    slice.offset(invalid_id);
+    return std::pair{slice.offset(), slice.offset() + slice.rows()};
   };
   auto g = [&](auto buffer) -> caf::error {
     // TODO: bind the lifetime of the table slice to the segment chunk. This
     // requires that table slices will be constructable from a chunk. Until
     // then, we stupidly deserialize the data into a new table slice.
-    table_slice_ptr slice;
-    if (auto err = unpack(*buffer->data_nested_root(), slice))
+    table_slice slice;
+    if (auto err = unpack(*buffer, slice))
```

Comment:
This needs to be changed to use `chunk_.slice()` with the appropriate offset / length pair and construct the `table_slice` from the returned `chunk_ptr`.